FRAuth/FRAuth/Authentication/NodeNext.swift

```diff
@@ -0,0 +1,224 @@
+                    }
+                }
+                else {
+                    completion(nil, node, error)
+                }
+            }
+        }
+    }
+    
+    func next(completion:@escaping NodeCompletion<Token>) {
+        
+        guard let thisRequest = try? self.buildAuthServiceRequest(),
+                let authServiceId = self.authServiceId,
+                let serverConfig = self.serverConfig,
+                let serviceName = self.serviceName,
+                let authIndexType = self.authIndexType
+        else { return }
+        
+        let thisAction = Action(type: .AUTHENTICATE, payload: ["tree": serviceName, "type": authIndexType])
+        
+        self.invoke(authServiceId: authServiceId, serverConfig: serverConfig, serviceName: serviceName, authIndexType: authIndexType, request: thisRequest, action: thisAction, completion: completion)
+    }
+    
+    func invoke(authServiceId: String, serverConfig: ServerConfig, serviceName: String, authIndexType: String, request: Request, action: Action, completion: @escaping NodeCompletion<Token>) {
+        
+        FRRestClient.invoke(request: request, action: action) { (result) in
+            switch result {
+            case .success(let response, _):
+                
+                // If authId received
+                if let _ = response[OpenAM.authId] {
+                    do {
+                        let node = try Node(authServiceId, response, serverConfig, serviceName, authIndexType, self.oAuth2Config, self.keychainManager, self.tokenManager)
+                        completion(nil, node, nil)
+                    } catch let authError as AuthError {
+                        completion(nil, nil, authError)
+                    } catch {
+                        completion(nil, nil, error)
+                    }
+                }
+                else if let tokenId = response[OpenAM.tokenId] as? String {
+                    let token = Token(tokenId)
+                    if let keychainManager = self.keychainManager {
+                        let currentSessionToken = keychainManager.getSSOToken()
+                        //If the `currentSessionToken` is nil and have OAuth2.0 tokens it means the user did a Centralized Login flow to authenticate initially. The token is now returned from either a Policy Advice and should be the same one as originaly created, or from a new authentication
+                        if ((currentSessionToken == nil) && (try? keychainManager.getAccessToken()) != nil) && authIndexType == "composite_advice" {
+                            // In this case we are running a transactional authorization flow, the new SSO Token is the same as the originally created one. When running Centralised login, this lived in the browser cookie storage and is unaccesssible from the app
+                            completion(token, nil, nil)
+                            return
+                        }
```

Comment:
Just to make sure I understand the change, lines 158-162 are the REAL CHANGE to fix the bug we were having right? The rest is just refactoring.